stan/math/fwd/scal/fun/ldexp.hpp

```diff
@@ -0,0 +1,27 @@
+#ifndef STAN_MATH_FWD_SCAL_FUN_LDEXP_HPP
+#define STAN_MATH_FWD_SCAL_FUN_LDEXP_HPP
+
+#include <stan/math/fwd/core.hpp>
+#include <stan/math/prim/scal/fun/ldexp.hpp>
+#include <cmath>
+
+namespace stan {
+namespace math {
+
+/**
+ * Returns the product of a (the significand) and
+ * 2 to power b (the exponent).
+ *
+ * @tparam T Scalar type of significand
+ * @param[in] a the significand
+ * @param[in] b an integer that is the exponent
+ * @return product of a times 2 to the power b
+ */
+template <typename T>
+inline fvar<T> ldexp(const fvar<T>& a, int b) {
+  return fvar<T>(ldexp(a.val_, b), ldexp(a.d_, b));
+}
```

Comment:
Should their be an overload for fvar<T> being the exponent?

Reply:
That's a good question. The ```std::ldexp(a,b)``` implementation only accepts integers for the exponent ([spec here](https://en.cppreference.com/w/cpp/numeric/math/ldexp)), and if a double is passed it gets internally cast to an int.

Given that the exponent needs to be an int, I don't think an ```fvar``` overload would work very well. Alternatively, we could just use the ```a * exp2(b)``` implementation, since that doesn't require ```b``` to be an integer.

According to the spec, it should also be faster as well:
> On many implementations, std::ldexp is less efficient than multiplication or division by a power of two using arithmetic operators.

Reply:
Right---we shouldn't allow `fvar<T>` or `var` in positions that require integers.  More generally, we shouldn't ever be casting `real` values to `int` through the Stan language.  That is, if there's a function `foo(int)` in the Stan language that requires its first argument to be an integer, we should not implement `foo(real)` and have it cast.